pennylane/transforms/optimization/pattern_matching.py

```diff
@@ -0,0 +1,1594 @@
+                        ):
+                            # Qubits mapping between circuit and pattern
+                            qubits_conf = _merge_first_match_and_permutation(
+                                first_match_qubits_conf, not_fixed_qubits_conf_permuted
+                            )
+
+                            # Update wires, target_wires, control_wires
+                            wires, target_wires, control_wires = _update_qubits(
+                                circuit_dag, qubits_conf
+                            )
+
+                            # Forward match part of the algorithm
+                            forward = ForwardMatch(
+                                circuit_dag,
+                                pattern_dag,
+                                node_c[0],
+                                node_p[0],
+                                wires,
+                                target_wires,
+                                control_wires,
+                            )
+                            forward.run_forward_match()
+
+                            # Backward match part of the algorithm
+                            backward = BackwardMatch(
+                                circuit_dag,
+                                pattern_dag,
+                                qubits_conf,
+                                forward.match,
+                                forward.circuit_matched_with,
+                                forward.circuit_blocked,
+                                forward.pattern_matched_with,
+                                node_c[0],
+                                node_p[0],
+                                wires,
+                                control_wires,
+                                target_wires,
+                            )
+                            backward.run_backward_match()
+
+                            _add_match(match_list, backward.match_final)
+
+    match_list.sort(key=lambda x: len(x.match), reverse=True)
+
+    # Extract maximal matches and optimizes the circuit for compatible maximal matches
+    if match_list:
+        maximal = MaximalMatches(match_list)
+        maximal.run_maximal_matches()
+        max_matches = maximal.max_match_list
+        return max_matches
+
+    return match_list
+
+
+def _compare_operation_without_qubits(node_1, node_2):
+    """Compare two operations without taking the qubits into account.
+
+    Args:
+        node_1 (.CommutationDAGNode): First operation.
+        node_2 (.CommutationDAGNode): Second operation.
+    Return:
+        Bool: True if similar operation (no qubits comparison) and False otherwise.
+    """
+    operation_1 = node_1.op
+    operation_2 = node_2.op
+    if operation_1.name == operation_2.name:
+        if operation_1.num_params == operation_2.num_params:
+            if operation_1.data == operation_2.data:
+                return True
```

Comment:
following up what jay said:
```suggestion
    op1 = node_1.op
    op2 = node_2.op
    return (op1.name == op2.name) and (op1.num_params == op2.num_params) and(op1.data == op1.data)
```